src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -863,13 +870,23 @@ public static object DeepCopyInner(object original)
             {
                 copy = copier(original);
                 SerializationContext.Current.RecordObject(original, copy);
+                return copy;
             }
-            else
+
+            var externalSerializer = externalSerializers.FirstOrDefault(serializer => serializer.IsSupportedType(t));
```

Comment:
If there's more than one custom serializer specified, we'll search them in the order of how they are listed in the config? With HashSet it's not clear to me that the order will be necessarily preserved. I think that another reason to consider using a list instead.

Do we actually envision multiple external serializers used at the same time? If not, then the code will be simpler with a single one. If we do need to support multiple, the `FirstOrDefault()` approach will be ambiguous if more than one external serializer support the same type because the precedence will be implicit.